src/Raven.Server/Documents/Indexes/IndexStore.cs

```diff
@@ -606,6 +611,10 @@ private Index HandleStaticIndexChange(string name, IndexDefinition definition)
                     return null;
                 }
 
+                var definitionClone = new IndexDefinition();
+                definition.CopyTo(definitionClone);
```

Comment:
this is error-prone, especially when merging between versions. We need a convention test for this where we need to perform copy to and assert if there are any changes. We also need to assert if we are setting all the fields to non-default values in the index definition in that test.

Reply:
Done